hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/request/validation/ValidatorRegistry.java

```diff
@@ -0,0 +1,278 @@
+   * - An EnumMap with the RequestType as the key, and in which
+   *   - values are an EnumMap with the request processing phase as the key, and in which
+   *     - values is an {@link IndexedItems } containing the validation list
+   * @param validatorsToBeRegistered collection of the annotated validtors to process.
+   */
+  private void initMaps(Class<RequestType> requestType,
+                        Set<RequestProcessingPhase> allowedPhases,
+                        Collection<Class<? extends Annotation>> validatorsToBeRegistered,
+                        Reflections reflections) {
+    for (Class<? extends Annotation> validator : validatorsToBeRegistered) {
+      registerValidator(requestType, allowedPhases, validator, reflections);
+    }
+  }
+
+
+
+  private void registerValidator(Class<RequestType> requestType,
+                                 Set<RequestProcessingPhase> allowedPhases,
+                                 Class<? extends Annotation> validatorToBeRegistered,
+                                 Reflections reflections) {
+    Collection<Method> methods =  reflections.getMethodsAnnotatedWith(validatorToBeRegistered);
+    Class<? extends Version> versionClass = (Class<? extends Version>)
+        this.getReturnTypeOfAnnotationMethod(validatorToBeRegistered, RegisterValidator.MAX_VERSION_METHOD_NAME);
+    List<Pair<? extends Annotation, Method>> sortedMethodsByMaxVersion = methods.stream()
+        .map(method -> Pair.of(method.getAnnotation(validatorToBeRegistered), method))
+        .sorted((validatorMethodPair1, validatorMethodPair2) ->
+            Integer.compare(
+                this.getMaxVersion(validatorMethodPair1.getKey()).getVersion(),
+                this.getMaxVersion(validatorMethodPair2.getKey()).getVersion()))
+        .collect(Collectors.toList());
+    for (Pair<? extends Annotation, Method> validatorMethodPair : sortedMethodsByMaxVersion) {
+      Annotation validator = validatorMethodPair.getKey();
+      Method method = validatorMethodPair.getValue();
+      Version maxVersion = this.getMaxVersion(validator);
+      RequestProcessingPhase phase = this.getRequestPhase(validator);
+      checkAllowedAnnotationValues(allowedPhases, phase, RegisterValidator.PROCESSING_PHASE_METHOD_NAME,
+          method.getName());
+      RequestType type = this.getRequestType(validator, requestType);
+      method.setAccessible(true);
+
+      EnumMap<RequestType, EnumMap<RequestProcessingPhase, IndexedItems<Method, Integer>>> requestMap =
+          this.indexedValidatorMap.get(versionClass);
+      EnumMap<RequestProcessingPhase, IndexedItems<Method, Integer>> phaseMap =
+          this.getAndInitialize(type, () -> new EnumMap<>(RequestProcessingPhase.class), requestMap);
+      this.getAndInitialize(phase, IndexedItems::new, phaseMap).add(method, maxVersion.getVersion());
+    }
+
+  }
+
+  private <K, V> V getAndInitialize(K key, Supplier<V> defaultSupplier, Map<K, V> from) {
```

Comment:
Probably we should just inline this, as it came up earlier when this method body was switched to computeIfAbsent.